backend/src/bundles/users/user.repository.ts

```diff
@@ -158,6 +168,36 @@ class UserRepository implements Repository {
     public delete(): ReturnType<Repository['delete']> {
         return Promise.resolve(true);
     }
+
+    public async findByReferralCode(
```

Comment:
You can refactor `find` method to join `user_details` table and find by query. This way this method will be reusable for every find case and you will not need the dedicated method to find by referral code